packages/adblocker/src/engine/bucket/cosmetic.ts

```diff
@@ -364,21 +368,36 @@ export default class CosmeticFilterBucket {
       return true;
     });
 
+    // Additional data for engine events
+    const matches: CosmeticFilter[] = [...rules];
```

Comment:
I don't see where we populate `matches` here.

Reply:
`rules` is never modified so you can return it as 
```
return {
  rules: modifications,
  matches: rules,
  exceptions,
};
```